crates/plugins/src/badger/store.rs

```diff
@@ -86,7 +86,7 @@ impl BadgerRecordManager {
         let new = BadgerRecord {
             name: name.to_owned(),
             badgered_from: from.clone(),
-            badgered_to: to.iter().map(|v| <semver::Version>::clone(v)).collect(),
+            badgered_to: to.iter().cloned().map(<semver::Version>::clone).collect(),
```

Comment:
Is the call to `map` still necessary?

Reply:
Oh, it could prove not to be! Thanks.  I admit I just did the barebones lint-suggested thing to get it to not error.

Reply:
I had the same thought yesterday but when i looked into it i started to believe that this is a false positive `map_clone` lint (=> https://github.com/rust-lang/rust-clippy/issues/12271). I.e. removing the `map` yields an type error on the resulting collection. 

A few things we could do but i'm not sure which is best:
1. Drop the `cloned` and attach `#[allow(clippy::map_clone)]` to this line
2. Keep as is and attach a note to remove the `cloned` whenever the false positive fix lands in rust proper (if it hasn't already and just requires a rust version bump 🤷‍♂️ ).